src/lib.rs

```diff
@@ -3496,14 +3512,8 @@ fn which(tool: &Path) -> Option<PathBuf> {
     })
 }
 
-/// Check if the file's extension is either "asm" or "s", case insensitive.
-fn is_asm(file: &Path) -> bool {
-    if let Some(ext) = file.extension() {
-        if let Some(ext) = ext.to_str() {
-            let ext = ext.to_lowercase();
-            return ext == "asm" || ext == "s";
-        }
-    }
-
-    false
+#[derive(Clone, Copy, PartialEq)]
+enum AsmExt {
+    S,
+    Asm,
 }
```

Comment:
Hm, if I didn't already know what the `asm_ext` function did then reading `asm_ext == Some(AsmExt::Asm)` above may feel like we're using a very redundantly named enum. Maybe the variants should be a bit more descriptive of their intended use? E.g. something like `AsmExt::Msvc` or `AsmExt::Masm` would seem clearer?

I don't feel particularly strongly about this and I do get the advantage of the variant names matching the tested extension.

Reply:
I've definitely seen `.S` used on MSVC-heavy projects, so I think that might be more confusing.

Reply:
Fair.

Reply:
I've renamed it to `AsmFileExt::DotAsm` which should be more clear.